PlantFriends.ino

```diff
@@ -501,22 +587,20 @@ void setup() {
   // Read battery voltage
   float voltage = (float)analogRead(A0) * 4.1f;
   Serial.print(F("Battery voltage: ")); Serial.println(voltage);
-
+  // Calculate state of charge in percent. This table is used for lipo voltage: https://blog.ampow.com/lipo-voltage-chart/
+  // The polynomial is fitted without the data points from 10% to 100% using this tool: https://arachnoid.com/polysolve/
+  float state_of_charge = -2964.08 +  (1369.59 * (voltage/1000.0)) - (152.366 * (voltage/1000.0) * (voltage/1000.0));
```

Comment:
Please append an `f` i.e. change `-2964.08` to `-2964.08f` to all the floating point numbers, so it's using single-precision instead of double-precision floating point math.

Reply:
Please also add a space before and after the `/` sign.